### PR TITLE
feat(playstation): Add event merging logic

### PR DESCRIPTION
**Pull request by tobias-wilfert**
Add logic that allows us to send an event from Tempest (containing all the info we only know there) along with the crash to Relay for processing. Previously all the processing needed to happen on the Tempest side and Relay would receive a "finished" event + attachments. 

This now enables us to greatly simplify Tempest and also have one spot where all the processing happens rather than two.

### CODE REVIEW

File: tests/integration/test_playstation.py [hunk 1]
```diff
@@ -451,3 +451,201 @@ def test_data_request(mini_sentry, relay_processing_with_playstation):
...
+        "breadcrumbs": {
+            "values": [
+                {
+                    "timestamp": time_within_delta(),
+                    "type": "default",
+                    "level": "info",
+                    "message": "crumb",
+                }
+            ]
+        },
```
Comment (vaind) [inline tests/integration/test_playstation.py]: Maybe another test case (or change this one) with `"breadcrumps": null`, just to make sure it works for sentry-native

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,6 +17,7 @@
 - Add new iPhone 17 devices. ([#5203](https://github.com/getsentry/relay/pull/5203))
 - Upgrade sqlparser and improve SQL parsing for span grouping. ([#5211](https://github.com/getsentry/relay/pull/5211))
 - Maps `unknown_error` span status to `internal_error` ([#5202](https://github.com/getsentry/relay/pull/5202))
+- Add event merging logic for Playstation crashes ([#5228](https://github.com/getsentry/relay/pull/5228))
```
Comment (jjbayer) [inline CHANGELOG.md]:
```suggestion
- Add event merging logic for Playstation crashes. ([#5228](https://github.com/getsentry/relay/pull/5228))
```

File: relay-server/src/services/processor/playstation.rs [hunk 3]
```diff
@@ -287,6 +282,31 @@ fn infer_content_type(filename: &str) -> ContentType {
...
+    item.set_payload(ContentType::Json, payload.to_owned());
+    item
+}
+
+fn merge_or_create_event_item(json: &str, event: Option<Item>) -> Item {
+    match event {
+        Some(event) => {
+            let event_json =
+                serde_json::from_slice::<serde_json::Value>(&event.payload()).unwrap_or_default();
+            let mut base_json = serde_json::from_str::<serde_json::Value>(json).unwrap_or_default();
```
Comment (jjbayer) [inline relay-server/src/services/processor/playstation.rs]: nit: not JSON once parsed.

```suggestion
            let event =
                serde_json::from_slice::<serde_json::Value>(&event.payload()).unwrap_or_default();
            let mut base_event = serde_json::from_str::<serde_json::Value>(json).unwrap_or_default();
```

File: relay-server/src/services/processor/playstation.rs [hunk 4]
```diff
@@ -287,6 +282,31 @@ fn infer_content_type(filename: &str) -> ContentType {
...
 }
 
+fn create_item_with_json_payload(payload: &str) -> Item {
+    let mut item = Item::new(ItemType::Event);
+    item.set_payload(ContentType::Json, payload.to_owned());
+    item
+}
+
+fn merge_or_create_event_item(json: &str, event: Option<Item>) -> Item {
+    match event {
```
Comment (jjbayer) [inline relay-server/src/services/processor/playstation.rs]: nit: Could do `let Some(event) = event else { return create_item_with_json_payload(json) };` here.

File: tests/integration/test_playstation.py [hunk 5]
```diff
@@ -451,3 +451,201 @@ def test_data_request(mini_sentry, relay_processing_with_playstation):
...
+    )
+
+    relay.send_envelope(PROJECT_ID, envelope)
+
+    outcomes = outcomes_consumer.get_outcomes()
+    assert len(outcomes) == 0
+
+    event, payload = attachments_consumer.get_event_only()
+    assert payload == {
+        "event_id": mock.ANY,
```
Comment (jjbayer) [inline tests/integration/test_playstation.py:513]: Does this mean the event ID is generated, not taken from `tempest_event`?

**Review by jjbayer — APPROVED** [review]

File: tests/integration/test_playstation.py (hunk 1 above)
Comment (tobias-wilfert) [reply tests/integration/test_playstation.py]: Updated the test to now use the dump that you shared with the `"breadcrumps": null`.
